library/src/main/java/com/getstream/sdk/chat/rest/core/Client.java

```diff
@@ -577,16 +586,28 @@ public synchronized void onSetUserCompleted(ClientConnectionCallback callback) {
 
     private synchronized void connect() {
         Log.i(TAG, "client.connect was called");
-        tokenProvider.getToken(userToken -> {
+
+        if (!anonymousConnection) {
+            tokenProvider.getToken(userToken -> {
+                try {
+                    webSocketService = webSocketServiceProvider.provideWebSocketService(getUser(), userToken, this);
+                    apiService = apiServiceProvider.provideApiService(tokenProvider);
+                    uploadStorage = uploadStorageProvider.provideUploadStorage(tokenProvider, this);
+                } catch (UnsupportedEncodingException e) {
+                    e.printStackTrace();
```

Comment:
why did you replace `onError` with ` e.printStackTrace();` sounds like a bad change